pkg/server/peer.go

```diff
@@ -282,8 +282,9 @@ func (peer *Peer) llgrFamilies() ([]bgp.RouteFamily, []bgp.RouteFamily) {
 
 func (peer *Peer) isLLGREnabledFamily(family bgp.RouteFamily) bool {
 	peer.fsm.lock.RLock()
-	defer peer.fsm.lock.RUnlock()
-	if !peer.fsm.pConf.GracefulRestart.Config.LongLivedEnabled {
+	llgrEnabled := peer.fsm.pConf.GracefulRestart.Config.LongLivedEnabled
+	peer.fsm.lock.RUnlock()
```

Comment:
There is significant precedent for this RLock/read config/RUnlock pattern in gobgp. This avoids the issue but leaves us open to inconsistent configuration because the two locks could read different states. Unfortunately, there is some precedent for that also in many other places in the code.

I think a better way to handle this would be to rearchitect the sharing of the fsm, fsm.pConf, and whatever else is under synchronization with this lock. I think that a single goroutine should be responsible for managing updates to fsm and should push copies of it to other goroutines through channels.

My observation is that updates to the fsm do not happen that frequently in the grand scheme of things. Also, I don't see any harm in another goroutine getting an update to the fsm a little bit late if it is in the middle of an operation. My opinion is that it would be more consistent for it to wait to get an update to the state by selecting on a channel.